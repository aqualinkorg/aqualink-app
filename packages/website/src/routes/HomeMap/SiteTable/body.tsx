import {
  Hidden,
  TableBody,
  TableCell,
  TableFooter,
  TablePagination,
  TableRow,
  Theme,
  Typography,
  useMediaQuery,
  useTheme,
} from '@mui/material';
import { WithStyles } from '@mui/styles';
import createStyles from '@mui/styles/createStyles';
import withStyles from '@mui/styles/withStyles';
import ErrorIcon from '@mui/icons-material/Error';
import React, { useEffect, useMemo, useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { TableRow as Row } from 'store/Homepage/types';
import { constructTableData } from 'store/Sites/helpers';
import { sitesToDisplayListSelector } from 'store/Sites/sitesListSlice';
import {
  siteOnMapSelector,
  setSiteOnMap,
  setSearchResult,
} from 'store/Homepage/homepageSlice';
import { dhwColorFinder } from 'helpers/degreeHeatingWeeks';
import { formatNumber } from 'helpers/numberUtils';
import { alertColorFinder } from 'helpers/bleachingAlertIntervals';
import { colors } from 'layout/App/theme';
import { calculateAdjustedLng } from 'helpers/map';
import L from 'leaflet';
import { getComparator, Order, OrderKeys, stableSort } from './utils';
import { Collection } from '../../Dashboard/collection';

const SCROLLT_TIMEOUT = 500;
const ROWS_PER_PAGE = 200;

const RowNameCell = ({
  site: { locationName, region },
  classes,
}: {
  site: Row;
  classes: SiteTableBodyProps['classes'];
}) => {
  return (
    <TableCell className={classes.nameCells}>
      <Typography align="left" variant="h6" color="textSecondary">
        {locationName}
      </Typography>

      {locationName !== region && region && (
        <Typography className={classes.regionName} variant="subtitle1">
          {region}
        </Typography>
      )}
    </TableCell>
  );
};

const RowNumberCell = ({
  color = colors.black,
  unit = '',
  decimalPlaces = 1,
  value,
  classes,
  isExtended = false,
}: {
  color?: string;
  unit?: string;
  value: number | null;
  decimalPlaces?: number;
  classes: SiteTableBodyProps['classes'];
  isExtended?: boolean;
}) => {
  return (
    <TableCell
      className={
        isExtended ? classes.cellTextAlignExtended : classes.cellTextAlign
      }
    >
      <Typography
        variant="h6"
        style={{ color }}
        className={classes.numberCellsTitle}
      >
        {formatNumber(value, decimalPlaces)}
        <Hidden smUp>
          &nbsp;
          <Typography variant="h6" component="span">
            {unit}
          </Typography>
        </Hidden>
      </Typography>
    </TableCell>
  );
};

const RowAlertCell = ({
  site: { alertLevel },
  classes,
}: {
  site: Row;
  classes: SiteTableBodyProps['classes'];
}) => {
  return (
    <TableCell className={classes.cellTextAlign}>
      <ErrorIcon
        style={{
          color: alertColorFinder(alertLevel),
        }}
      />
    </TableCell>
  );
};

const SiteTableBody = ({
  order,
  orderBy,
  isExtended = false,
  collection,
  scrollTableOnSelection = true,
  scrollPageOnSelection = false,
  map,
  classes,
  onEmptyColumnsChange,
}: SiteTableBodyProps) => {
  const dispatch = useDispatch();
  const storedSites = useSelector(sitesToDisplayListSelector);
  const sitesList = useMemo(
    () => collection?.sites || storedSites || [],
    [collection, storedSites],
  );
  const siteOnMap = useSelector(siteOnMapSelector);
  const [selectedRow, setSelectedRow] = useState<number>();
  const [page, setPage] = useState(0);

  const theme = useTheme();
  const isTablet = useMediaQuery(theme.breakpoints.down('md'));

  const tableData = useMemo(
    () =>
      stableSort<Row>(
        constructTableData(sitesList),
        getComparator(order, orderBy),
      ),
    [order, orderBy, sitesList],
  );

  // Check which columns have all empty values
  const emptyColumns = useMemo(() => {
    const columnsToCheck = [OrderKeys.BUOY_TOP, OrderKeys.BUOY_BOTTOM];

    return columnsToCheck.reduce((emptyCols, column) => {
      const hasValue = tableData.some((site) => {
        if (column === OrderKeys.BUOY_TOP) return site.buoyTop !== null;
        if (column === OrderKeys.BUOY_BOTTOM) return site.buoyBottom !== null;
        return false;
      });

      if (!hasValue) {
        // eslint-disable-next-line fp/no-mutating-methods
        emptyCols.push(column);
      }
      return emptyCols;
    }, [] as OrderKeys[]);
  }, [tableData]);

  // Notify parent component of empty columns
  useEffect(() => {
    onEmptyColumnsChange?.(emptyColumns);
  }, [emptyColumns, onEmptyColumnsChange]);

  const idToIndexMap = useMemo(
    () =>
      tableData.reduce((acc, item, index) => {
        // eslint-disable-next-line fp/no-mutation
        acc[item.tableData.id] = index;
        return acc;
      }, {} as Record<number, number>),
    [tableData],
  );

  const handleClick = (event: unknown, site: Row) => {
    setSelectedRow(site.tableData.id);
    dispatch(setSearchResult());

    const targetSite = sitesList[site.tableData.id];
    if (!targetSite || targetSite.polygon.type !== 'Point') return;
    const [lng] = targetSite.polygon.coordinates;
    const adjustedLng = calculateAdjustedLng(map || null, lng);
    dispatch(setSiteOnMap({ ...targetSite, displayLng: adjustedLng }));

    const mapElement = document.getElementById('sites-map');
    if (scrollPageOnSelection && mapElement) {
      mapElement.scrollIntoView({ block: 'center', behavior: 'smooth' });
    }
  };

  useEffect(() => {
    const index = sitesList.findIndex((item) => item.id === siteOnMap?.id);
    setSelectedRow(index);
  }, [siteOnMap, sitesList]);

  // scroll to the relevant site row when site is selected.
  useEffect(() => {
    if (selectedRow === undefined || idToIndexMap[selectedRow] === undefined)
      return;
    // Go to the page where the selected site is
    setPage(Math.floor(idToIndexMap[selectedRow] / ROWS_PER_PAGE));
    // only scroll if not on mobile (info at the top is more useful than the site row)
    if (!isTablet && scrollTableOnSelection) {
      setTimeout(() => {
        const child = document.getElementById(
          `homepage-table-row-${selectedRow}`,
        );

        child?.scrollIntoView({ block: 'center', behavior: 'smooth' });
      }, SCROLLT_TIMEOUT);
    }
  }, [idToIndexMap, isTablet, scrollTableOnSelection, selectedRow]);

  // Handle page change
  const handlePageChange = (event: unknown, newPage: number) => {
    setPage(newPage);
  };

  return (
    <>
      <TableBody>
        {tableData
          .slice(page * ROWS_PER_PAGE, page * ROWS_PER_PAGE + ROWS_PER_PAGE)
          .map((site) => {
            return (
              <TableRow
                id={`homepage-table-row-${site.tableData.id}`}
                hover
                className={classes.tableRow}
                style={{
                  backgroundColor:
                    site.tableData.id === selectedRow
                      ? colors.lighterBlue
                      : 'white',
                }}
                onClick={(event) => handleClick(event, site)}
                role="button"
                tabIndex={-1}
                key={site.tableData.id}
              >
                <RowNameCell
                  site={site}
                  classes={{
                    ...classes,
                    nameCells: isExtended
                      ? classes.extendedTableNameCells
                      : classes.nameCells,
                  }}
                />
                <RowNumberCell
                  isExtended={isExtended}
                  classes={classes}
                  value={site.sst}
                  color={isExtended ? colors.black : colors.lightBlue}
                  unit="°C"
                />
                {isExtended && (
                  <RowNumberCell
                    isExtended={isExtended}
                    classes={classes}
                    value={site.historicMax}
                    color={colors.black}
                    unit="°C"
                  />
                )}
                {isExtended && (
                  <RowNumberCell
                    isExtended={isExtended}
                    classes={classes}
                    value={site.sstAnomaly}
                    color={colors.black}
                    unit="°C"
                  />
                )}
                <RowNumberCell
                  isExtended={isExtended}
                  classes={classes}
                  value={site.dhw}
                  color={dhwColorFinder(site.dhw)}
                  unit="DHW"
                />
                {isExtended && !emptyColumns.includes(OrderKeys.BUOY_TOP) && (
                  <RowNumberCell
                    isExtended={isExtended}
                    classes={classes}
                    value={site.buoyTop}
                    color={colors.black}
                    unit="°C"
                  />
                )}
                {isExtended &&
                  !emptyColumns.includes(OrderKeys.BUOY_BOTTOM) && (
                    <RowNumberCell
                      isExtended={isExtended}
                      classes={classes}
                      value={site.buoyBottom}
                      color={colors.black}
                      unit="°C"
                    />
                  )}
                <RowAlertCell site={site} classes={classes} />
              </TableRow>
            );
          })}
      </TableBody>
      {sitesList.length > ROWS_PER_PAGE && (
        <TableFooter>
          <TableRow>
            <TablePagination
              className={classes.stickyFooter}
              count={sitesList.length}
              rowsPerPage={ROWS_PER_PAGE}
              rowsPerPageOptions={[ROWS_PER_PAGE]}
              page={page}
              onPageChange={handlePageChange}
              slotProps={{
                displayedRows: {
                  sx: { marginBottom: 0 },
                },
              }}
            />
          </TableRow>
        </TableFooter>
      )}
    </>
  );
};

const styles = (theme: Theme) =>
  createStyles({
    nameCells: {
      paddingLeft: 10,
      maxWidth: 120,
      overflowWrap: 'break-word',
      [theme.breakpoints.down('sm')]: { width: '35%', paddingRight: 0 },
    },
    extendedTableNameCells: {
      paddingLeft: 10,
      maxWidth: 200,
      overflowWrap: 'break-word',
      [theme.breakpoints.down('sm')]: { width: '10%', paddingRight: 0 },
    },
    regionName: {
      color: 'gray',
    },
    numberCellsTitle: {
      [theme.breakpoints.down('sm')]: { fontWeight: 600 },
    },
    cellTextAlign: {
      textAlign: 'left',
      [theme.breakpoints.down('sm')]: {
        textAlign: 'right',
      },
    },
    cellTextAlignExtended: {
      [theme.breakpoints.down('sm')]: {
        paddingLeft: 10,
        paddingRight: 0,
        textAlign: 'left',
      },
    },
    tableRow: {
      cursor: 'pointer',
      borderTop: `1px solid ${theme.palette.grey['300']}`,
    },
    stickyFooter: {
      position: 'sticky',
      bottom: 0,
    },
  });

interface SiteTableBodyIncomingProps {
  order: Order;
  orderBy: OrderKeys;
  isExtended?: boolean;
  collection?: Collection;
  scrollTableOnSelection?: boolean;
  scrollPageOnSelection?: boolean;
  map?: L.Map | null;
  onEmptyColumnsChange: (emptyColumns: OrderKeys[]) => void;
}

type SiteTableBodyProps = WithStyles<typeof styles> &
  SiteTableBodyIncomingProps;

export default withStyles(styles)(SiteTableBody);
