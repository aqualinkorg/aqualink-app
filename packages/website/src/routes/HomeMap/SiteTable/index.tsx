import React, { ReactNode, useState } from 'react';
import { times } from 'lodash';
import {
  Box,
  CircularProgress,
  Hidden,
  MenuItem,
  Select,
  SelectProps,
  Table,
  TableContainer,
  Theme,
  Typography,
} from '@mui/material';
import { WithStyles } from '@mui/styles';
import createStyles from '@mui/styles/createStyles';
import withStyles from '@mui/styles/withStyles';
import { useSelector } from 'react-redux';
import classNames from 'classnames';
import { ArrowDownward, ArrowUpward } from '@mui/icons-material';
import L from 'leaflet';

import { siteOnMapSelector } from 'store/Homepage/homepageSlice';
import { getSiteNameAndRegion } from 'store/Sites/helpers';
import { useWindowSize } from 'hooks/useWindowSize';
import { sitesListLoadingSelector } from 'store/Sites/sitesListSlice';
import SelectedSiteCard from './SelectedSiteCard';
import SiteTableBody from './body';
import { getOrderKeysFriendlyString, Order, OrderKeys } from './utils';
import EnhancedTableHead from './tableHead';
import { Collection } from '../../Dashboard/collection';
import { SitesFilterModal } from './SitesFilterDropdown';

const SMALL_HEIGHT = 720;

const DEFAULT_ITEMS: OrderKeys[] = [
  OrderKeys.ALERT,
  OrderKeys.DHW,
  OrderKeys.LOCATION_NAME,
  OrderKeys.SST,
];

const MOBILE_SELECT_MENU_ITEMS = Object.values(OrderKeys)
  .filter((key) => DEFAULT_ITEMS.includes(key))
  .reduce<ReactNode[]>(
    (elements, val) => [
      ...elements,
      ...times(2, (i) => {
        const itemOrder: Order = i % 2 === 0 ? 'asc' : 'desc';
        return (
          <MenuItem value={`${val}-${itemOrder}`} key={val + i}>
            <Typography color="primary" variant="h4">
              {getOrderKeysFriendlyString(val)}
              {'  '}
              {itemOrder === 'asc' ? (
                <ArrowDownward fontSize="small" />
              ) : (
                <ArrowUpward fontSize="small" />
              )}
            </Typography>
          </MenuItem>
        );
      }),
    ],
    [],
  );

const SiteTable = ({
  isDrawerOpen = false,
  showCard = true,
  showSiteFiltersDropdown = true,
  isExtended = false,
  collection,
  scrollTableOnSelection = true,
  scrollPageOnSelection,
  map,
  classes,
}: SiteTableProps) => {
  const loading = useSelector(sitesListLoadingSelector);
  const siteOnMap = useSelector(siteOnMapSelector);
  const { height } = useWindowSize() || {};

  const [order, setOrder] = useState<Order>('desc');
  const [orderBy, setOrderBy] = useState<OrderKeys>(OrderKeys.ALERT);

  const handleRequestSort = (event: unknown, property: OrderKeys) => {
    const isAsc = orderBy === property && order === 'asc';
    setOrder(isAsc ? 'desc' : 'asc');
    setOrderBy(property);
  };

  // This function is used to prevent the drawer onClick close effect on mobile
  const onInteractiveClick = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>,
  ) => {
    event.stopPropagation();
  };

  const onMobileSelectChange: SelectProps['onChange'] = (newValue) => {
    const value = newValue.target.value as string;
    const [newOrderBy, newOrder] = value.split('-') as [OrderKeys, Order];
    setOrder(newOrder);
    setOrderBy(newOrderBy);
  };

  // Get empty columns from SiteTableBody
  const [emptyColumns, setEmptyColumns] = useState<OrderKeys[]>([]);

  return (
    <>
      {/* Holds drawer handle and site name text on mobile */}
      {showCard && (
        <Hidden mdUp>
          <Box
            width="100vw"
            display="flex"
            justifyContent="center"
            marginTop={2}
            marginBottom={3}
          >
            <Box
              className={classNames(classes.topHandle, {
                [classes.bounce]: !!siteOnMap && !isDrawerOpen,
              })}
            />
            {!isDrawerOpen && (
              <Typography
                className={classes.allSitesText}
                variant="h5"
                color="textSecondary"
              >
                {siteOnMap ? getSiteNameAndRegion(siteOnMap).name : 'All Sites'}
              </Typography>
            )}
          </Box>
        </Hidden>
      )}
      {showCard && <SelectedSiteCard />}
      {showSiteFiltersDropdown && (
        <Box className={classes.dropdownWrapper}>
          <SitesFilterModal />
        </Box>
      )}
      {/* Holds sort selector on mobile. Sorting on desktop uses table headers. */}
      {!isExtended && (
        <Hidden mdUp>
          <Box
            paddingX={2}
            paddingY={3}
            display="flex"
            alignItems="center"
            onClick={onInteractiveClick}
          >
            <Typography variant="h5">Sort By: </Typography>
            <Select
              variant="standard"
              value={`${orderBy}-${order}`}
              className={classes.mobileSortSelect}
              onChange={onMobileSelectChange}
            >
              {MOBILE_SELECT_MENU_ITEMS}
            </Select>
          </Box>
        </Hidden>
      )}
      <Box
        className={
          height && height > SMALL_HEIGHT
            ? `${classes.tableHolder} ${classes.scrollable}`
            : `${classes.tableHolder}`
        }
        display="flex"
        flexDirection="column"
        flex={1}
      >
        <TableContainer>
          <Table
            stickyHeader
            className={isExtended ? classes.extendedTable : classes.table}
          >
            <Hidden mdDown={!isExtended}>
              <EnhancedTableHead
                order={order}
                orderBy={orderBy}
                onRequestSort={handleRequestSort}
                isExtended={isExtended}
                emptyColumns={emptyColumns}
              />
            </Hidden>
            <SiteTableBody
              order={order}
              orderBy={orderBy}
              isExtended={isExtended}
              collection={collection}
              scrollTableOnSelection={scrollTableOnSelection}
              scrollPageOnSelection={scrollPageOnSelection}
              map={map}
              onEmptyColumnsChange={setEmptyColumns}
            />
          </Table>
        </TableContainer>
        {loading && (
          <Box
            display="flex"
            flex={1}
            alignItems="center"
            justifyContent="center"
          >
            <CircularProgress size="4rem" thickness={1} />
          </Box>
        )}
      </Box>
    </>
  );
};
const styles = (theme: Theme) =>
  createStyles({
    tableHolder: {
      paddingLeft: 10,
      [theme.breakpoints.down('sm')]: {
        paddingLeft: 0,
        height: 'auto',
      },
    },
    scrollable: {
      overflowY: 'auto',
    },
    table: {
      borderCollapse: 'collapse',
    },
    extendedTable: {
      [theme.breakpoints.down('sm')]: {
        minWidth: 1220,
      },
      borderCollapse: 'collapse',
    },
    dropdownWrapper: {
      padding: '0 0 10px 10px',
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-start',
    },
    topHandle: {
      width: 50,
      height: 10,
      backgroundColor: theme.palette.grey['400'],
      borderRadius: '20px',
    },
    mobileSortSelect: {
      marginLeft: theme.spacing(2),
    },
    allSitesText: {
      position: 'absolute',
      left: 25,
      top: 25,
      whiteSpace: 'nowrap',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      maxWidth: '90vw',
    },
    bounce: { animation: '$bounce 1s infinite alternate' },
    '@keyframes bounce': {
      '0%': { transform: 'translateY(0px)' },
      '100%': { transform: 'translateY(-5px)' },
    },
  });

interface SiteTableProps
  extends SiteTableIncomingProps,
    WithStyles<typeof styles> {}

interface SiteTableIncomingProps {
  // used on mobile to add descriptive elements if the drawer is closed.
  isDrawerOpen?: boolean;
  showCard?: boolean;
  showSiteFiltersDropdown?: boolean;
  isExtended?: boolean; // Determines whether an extended version of the table will be displayed or not
  collection?: Collection;
  scrollTableOnSelection?: boolean;
  scrollPageOnSelection?: boolean;
  map?: L.Map | null; // Add map prop type
}

export default withStyles(styles)(SiteTable);
