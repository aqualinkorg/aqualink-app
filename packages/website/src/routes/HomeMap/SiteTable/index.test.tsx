import React from 'react';
import configureStore from 'redux-mock-store';

import { mockSite } from 'mocks/mockSite';
import { mockUser } from 'mocks/mockUser';
import { renderWithProviders } from 'utils/test-utils';
import SiteTable from '.';

vi.mock('./SelectedSiteCard', () => ({ default: 'Mock-SelectedSiteCard' }));

const mockStore = configureStore([]);

describe('SiteTable', () => {
  let element: HTMLElement;
  beforeEach(() => {
    const store = mockStore({
      user: {
        userInfo: mockUser,
      },
      sitesList: {
        list: [mockSite],
        loading: false,
        error: null,
        filters: {},
      },
      selectedSite: {
        loading: false,
        error: null,
      },
      homepage: {
        siteOnMap: null,
      },
    });

    const openDrawer = false;

    store.dispatch = vi.fn();

    element = renderWithProviders(<SiteTable isDrawerOpen={openDrawer} />, {
      store,
    }).container;
  });

  it('should render with given state from Redux store', () => {
    expect(element).toMatchSnapshot();
  });
});
