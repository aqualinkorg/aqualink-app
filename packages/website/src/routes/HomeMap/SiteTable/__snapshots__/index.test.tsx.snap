// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteTable should render with given state from Redux store 1`] = `
<div>
  <mock-hidden
    mdup="true"
  >
    <mock-box
      display="flex"
      justifycontent="center"
      marginbottom="3"
      margintop="2"
      width="100vw"
    >
      <mock-box
        classname="SiteTable-topHandle-6"
      />
      <mock-typography
        classname="SiteTable-allSitesText-8"
        color="textSecondary"
        variant="h5"
      >
        All Sites
      </mock-typography>
    </mock-box>
  </mock-hidden>
  <mock-selectedsitecard />
  <mock-box
    classname="SiteTable-dropdownWrapper-5"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInputBase-colorPrimary css-12kco1g-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      style="background-color: rgb(70, 154, 187); border-radius: 4px;"
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1iiz54r-MuiSelect-select-MuiInputBase-input-MuiInput-input"
        role="combobox"
        tabindex="0"
      >
        <span
          aria-hidden="true"
          class="notranslate"
        >
          â€‹
        </span>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-1gpuzre-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <mock-popover
        anchorel="[object HTMLDivElement]"
        anchororigin="[object Object]"
        classname="css-s4soc1-MuiMenu-root"
        id="menu-"
        open="false"
        slotprops="[object Object]"
        slots="[object Object]"
        transformorigin="[object Object]"
        transitionduration="auto"
        transitionprops="[object Object]"
      >
        <mock-menulist
          actions="[object Object]"
          autofocusitem="false"
          classname="MuiMenu-list css-7drlel-MuiMenu-list"
          disablelistwrap="true"
          id=":r0:"
          role="listbox"
          variant="selectedMenu"
        >
          <mock-menuitem
            aria-selected="false"
            data-value="All sites"
            role="option"
            selected="false"
          >
            <mock-typography
              style="color: black;"
            >
              All sites
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            aria-selected="false"
            data-value="Active buoys"
            role="option"
            selected="false"
          >
            <mock-typography
              style="color: black;"
            >
              Active buoys
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            aria-selected="false"
            data-value="Live streams"
            role="option"
            selected="false"
          >
            <mock-typography
              style="color: black;"
            >
              Live streams
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            aria-selected="false"
            data-value="3D Models"
            role="option"
            selected="false"
          >
            <mock-typography
              style="color: black;"
            >
              3D Models
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            aria-selected="false"
            data-value="HOBO loggers"
            role="option"
            selected="false"
          >
            <mock-typography
              style="color: black;"
            >
              HOBO loggers
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            aria-selected="false"
            data-value="Water quality"
            role="option"
            selected="false"
          >
            <mock-typography
              style="color: black;"
            >
              Water quality
            </mock-typography>
          </mock-menuitem>
          <mock-menuitem
            aria-selected="false"
            data-value="Reef Check"
            role="option"
            selected="false"
          >
            <mock-typography
              style="color: black;"
            >
              Reef Check
            </mock-typography>
          </mock-menuitem>
        </mock-menulist>
      </mock-popover>
    </div>
  </mock-box>
  <mock-hidden
    mdup="true"
  >
    <mock-box
      alignitems="center"
      display="flex"
      paddingx="2"
      paddingy="3"
    >
      <mock-typography
        variant="h5"
      >
        Sort By: 
      </mock-typography>
      <div
        class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary SiteTable-mobileSortSelect-7 css-r2nq11-MuiInputBase-root-MuiInput-root-MuiSelect-root"
      >
        <div
          aria-controls=":r1:"
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-standard MuiInputBase-input MuiInput-input css-1iiz54r-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          role="combobox"
          tabindex="0"
        >
          <mock-typography
            color="primary"
            variant="h4"
          >
            Alert
              
            <mock-arrowupward
              fontsize="small"
            />
          </mock-typography>
        </div>
        <input
          aria-hidden="true"
          aria-invalid="false"
          class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
          tabindex="-1"
          value="alert-desc"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-1gpuzre-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <mock-popover
          anchorel="[object HTMLDivElement]"
          anchororigin="[object Object]"
          classname="css-s4soc1-MuiMenu-root"
          id="menu-"
          open="false"
          slotprops="[object Object]"
          slots="[object Object]"
          transformorigin="[object Object]"
          transitionduration="auto"
          transitionprops="[object Object]"
        >
          <mock-menulist
            actions="[object Object]"
            autofocusitem="false"
            classname="MuiMenu-list css-7drlel-MuiMenu-list"
            disablelistwrap="true"
            id=":r1:"
            role="listbox"
            variant="selectedMenu"
          >
            <mock-menuitem
              aria-selected="false"
              data-value="alert-asc"
              role="option"
              selected="false"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                Alert
                  
                <mock-arrowdownward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem
              aria-selected="true"
              data-value="alert-desc"
              role="option"
              selected="true"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                Alert
                  
                <mock-arrowupward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem
              aria-selected="false"
              data-value="dhw-asc"
              role="option"
              selected="false"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                DHW
                  
                <mock-arrowdownward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem
              aria-selected="false"
              data-value="dhw-desc"
              role="option"
              selected="false"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                DHW
                  
                <mock-arrowupward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem
              aria-selected="false"
              data-value="locationName-asc"
              role="option"
              selected="false"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                Location Name
                  
                <mock-arrowdownward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem
              aria-selected="false"
              data-value="locationName-desc"
              role="option"
              selected="false"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                Location Name
                  
                <mock-arrowupward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem
              aria-selected="false"
              data-value="sst-asc"
              role="option"
              selected="false"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                Sst
                  
                <mock-arrowdownward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
            <mock-menuitem
              aria-selected="false"
              data-value="sst-desc"
              role="option"
              selected="false"
            >
              <mock-typography
                color="primary"
                variant="h4"
              >
                Sst
                  
                <mock-arrowupward
                  fontsize="small"
                />
              </mock-typography>
            </mock-menuitem>
          </mock-menulist>
        </mock-popover>
      </div>
    </mock-box>
  </mock-hidden>
  <mock-box
    classname="SiteTable-tableHolder-1 SiteTable-scrollable-2"
    display="flex"
    flex="1"
    flexdirection="column"
  >
    <mock-tablecontainer>
      <mock-table
        classname="SiteTable-table-3"
        stickyheader="true"
      >
        <mock-hidden
          mddown="true"
        >
          <mock-tablehead
            style="background-color: rgb(244, 244, 244);"
          >
            <mock-tablerow>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-11"
                padding="normal"
                sortdirection="false"
                style="width: 40%;"
              >
                <mock-tablesortlabel
                  active="false"
                  direction="asc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    SITE
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-11"
                padding="normal"
                sortdirection="false"
              >
                <mock-tablesortlabel
                  active="false"
                  direction="asc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    SST
                    <mock-typography
                      component="span"
                      style="color: black;"
                      variant="subtitle2"
                    >
                       (Â°C)
                    </mock-typography>
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-11"
                padding="normal"
                sortdirection="false"
              >
                <mock-tablesortlabel
                  active="false"
                  direction="asc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    STRESS
                    <mock-typography
                      component="span"
                      style="color: black;"
                      variant="subtitle2"
                    >
                       (DHW)
                    </mock-typography>
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
              <mock-tablecell
                align="left"
                classname="EnhancedTableHead-headCells-11"
                padding="normal"
                sortdirection="desc"
                style="width: 5%;"
              >
                <mock-tablesortlabel
                  active="true"
                  direction="desc"
                >
                  <mock-typography
                    nowrap="true"
                    style="color: black;"
                    variant="h6"
                  >
                    ALERT
                  </mock-typography>
                </mock-tablesortlabel>
              </mock-tablecell>
            </mock-tablerow>
          </mock-tablehead>
        </mock-hidden>
        <mock-tablebody>
          <mock-tablerow
            classname="SiteTableBody-tableRow-18"
            hover="true"
            id="homepage-table-row-0"
            role="button"
            style="background-color: white;"
            tabindex="-1"
          >
            <mock-tablecell
              classname="SiteTableBody-nameCells-12"
            >
              <mock-typography
                align="left"
                color="textSecondary"
                variant="h6"
              >
                Hawaii
              </mock-typography>
            </mock-tablecell>
            <mock-tablecell
              classname="SiteTableBody-cellTextAlign-16"
            >
              <mock-typography
                classname="SiteTableBody-numberCellsTitle-15"
                style="color: rgb(22, 141, 189);"
                variant="h6"
              >
                15.9
                <mock-hidden
                  smup="true"
                >
                  Â 
                  <mock-typography
                    component="span"
                    variant="h6"
                  >
                    Â°C
                  </mock-typography>
                </mock-hidden>
              </mock-typography>
            </mock-tablecell>
            <mock-tablecell
              classname="SiteTableBody-cellTextAlign-16"
            >
              <mock-typography
                classname="SiteTableBody-numberCellsTitle-15"
                style="color: rgb(8, 91, 163);"
                variant="h6"
              >
                0.0
                <mock-hidden
                  smup="true"
                >
                  Â 
                  <mock-typography
                    component="span"
                    variant="h6"
                  >
                    DHW
                  </mock-typography>
                </mock-hidden>
              </mock-typography>
            </mock-tablecell>
            <mock-tablecell
              classname="SiteTableBody-cellTextAlign-16"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ErrorIcon"
                focusable="false"
                style="color: rgb(198, 229, 250);"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-2h2zm0-4h-2V7h2z"
                />
              </svg>
            </mock-tablecell>
          </mock-tablerow>
        </mock-tablebody>
      </mock-table>
    </mock-tablecontainer>
  </mock-box>
</div>
`;
